src/psyclone/psyir/nodes/scoping_node.py

```diff
@@ -119,11 +125,57 @@ def _refine_copy(self, other):
         :type other: :py:class:`psyclone.psyir.node.Node`
 
         '''
-        super(ScopingNode, self)._refine_copy(other)
+        # Reorganise the symbol table construction to occur before we add
+        # the children.
         self._symbol_table = other.symbol_table.deep_copy()
-        # pylint: disable-next=protected-access
         self._symbol_table._node = self  # Associate to self
 
+        # Remove symbols corresponding to Routines that are contained in this
+        # ScopingNode. These symbols will be added automatically by the Routine
+        # objects when we add them to our child list in the super refine_copy
+        # call.
+        # We have to import Routine here to avoid a circular dependency.
+        # pylint: disable=import-outside-toplevel
+        from psyclone.psyir.nodes.routine import Routine
+        for node in other.walk(Routine):
+            try:
+                if isinstance(self._symbol_table.lookup(node.name),
+                              RoutineSymbol):
+                    self._symbol_table.remove(
+                            self._symbol_table.lookup(node.name))
+            except KeyError:
+                pass
+            except ValueError:
+                # If remove fails for the RoutineSymbol then we have to force
+                # the removal of the symbol.
+                symbol = self._symbol_table.lookup(node.name)
+                norm_name = self._symbol_table._normalize(symbol.name)
+                self._symbol_table._symbols.pop(norm_name)
+
+        super(ScopingNode, self)._refine_copy(other)
+        # pylint: disable=protected-access
+
+        # Update of children references to point to the equivalent symbols in
+        # the new symbol table attached to self.
+        # TODO #1377 Unfortunately Loop nodes currently store the associated
+        # loop variable in a `_variable` property rather than as a child so we
+        # must handle those separately. Also, in the LFRic API a Loop does not
+        # initially have the `_variable` property set which means that calling
+        # the `variable` getter causes an error (because it checks the
+        # internal-consistency of the Loop node). We therefore have to check
+        # the value of the 'private' `_variable` for now.
+        # We have to import Loop here to avoid a circular dependency.
+        # pylint: disable=import-outside-toplevel
+        from psyclone.psyir.nodes.loop import Loop
+        for node in self.walk((Reference, Loop)):
+            if isinstance(node, Reference):
+                if node.symbol in other.symbol_table.symbols:
+                    node.symbol = self.symbol_table.lookup(node.symbol.name)
+            if isinstance(node, Loop) and node._variable:
+                if node.variable in other.symbol_table.symbols:
+                    node.variable = self.symbol_table.lookup(
+                        node.variable.name)
```

Comment:
Is this not the same that the `replace_symbols_using` is doing just below?

Reply:
I think perhaps I broke `replace_symbols_using` and all of this code somehow when attempting to fix conflicts - I'll check master.

Reply:
Nvm I misunderstood how that function worked so the code above can be removed.